libopenage/util/compiler.h

```diff
@@ -8,25 +8,55 @@
  *
  * May contain some platform-dependent code.
  */
+ #include <ciso646>
 
 // pxd: from libcpp.string cimport string
 #include <string>
 
+/**
+ * DLL entry-point decorations.
+ */
+#if defined(_WIN32)
+	#if defined(libopenage_EXPORTS)
+		#define LIBOPENAGE_API __declspec(dllexport)
+	#else
+		#define LIBOPENAGE_API __declspec(dllimport)
+	#endif /* libopenage_EXPORTS */
+#else
+	#define LIBOPENAGE_API
+#endif
+
+#if defined(_MSC_VER)
+#ifndef HAVE_SSIZE_T
+typedef int ssize_t;
+#define HAVE_SSIZE_T 1
+#endif
+#endif
+
 /*
  * Branch prediction tuning.
  * the expression is expected to be true (=likely) or false (=unlikely).
  *
  * btw, this implementation was taken from the Linux kernel.
  */
+#if defined(__GNUC__)
 #define likely(x)    __builtin_expect(!!(x), 1)
 #define unlikely(x)  __builtin_expect(!!(x), 0)
+#else
+#define likely(x)   (x)
+#define unlikely(x) (x)
```

Comment:
I actually took it from a cython generated cpp file:
```
/* Test for GCC > 2.95 */
#if defined(__GNUC__)     && (__GNUC__ > 2 || (__GNUC__ == 2 && (__GNUC_MINOR__ > 95)))
  #define likely(x)   __builtin_expect(!!(x), 1)
  #define unlikely(x) __builtin_expect(!!(x), 0)
#else /* !__GNUC__ or GCC < 2.95 */
  #define likely(x)   (x)
  #define unlikely(x) (x)
#endif /* __GNUC__ */
```
I'm not aware of a branch prediction tuning feature for MSVC, if one exists.